openapi_python_client/parser/properties.py

```diff
@@ -63,24 +70,42 @@ def get_imports(self, *, prefix: str) -> Set[str]:
             prefix: A prefix to put before any relative (local) module names. This should be the number of . to get
             back to the root of the generated client.
         """
-        if self.nullable or not self.required:
+        if self.required and self.nullable:
             return {"from typing import Optional"}
+        elif not self.required:
+            return {"from typing import Union",
+                    f"from {prefix}types import Unset",
+                    f"from {prefix}types import UNSET"}
         return set()
 
     def to_string(self) -> str:
         """ How this should be declared in a dataclass """
         if self.default:
             default = self.default
         elif not self.required:
-            default = "None"
+            default = "UNSET"
         else:
             default = None
 
         if default is not None:
-            return f"{self.python_name}: {self.get_type_string()} = {self.default}"
+            return f"{self.python_name}: {self.get_type_string()} = {default}"
         else:
             return f"{self.python_name}: {self.get_type_string()}"
 
+    def to_query_string(self) -> str:
```

Comment:
This threw me for a while - I thought `get_query_string` was building literal query strings, e,g. `?property=value` which this is... not


I think you should rename this to something like

```suggestion
    def to_query_method_arg(self) -> str:
```

or
```suggestion
    def to_query_parameter(self) -> str:
```

(and `get_query_string()`/`query_string()` to match)

Reply:
Thanks, I renamed it. It's unclear to me whether the "new" query behavior should be the default the this response body/model body should be the new method, but it seems to me like it's used in more places.

I also had already changed it and then had to put back the query parsing....